{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- fine-annoation with CNV inference with `infercnv`, `copykat`, `scevan`\n",
    "- run in R 4.3.3 under conda env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import scanpy as sc\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import infercnvpy as cnv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('../output/annotated.h5ad')\n",
    "adata.obs = adata.obs[['sample_id', 'sample_info', 'age', 'sex', 'barcode', 'cell_type', 'grade', 'GFP', ]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_name', 'level', 'mgi_id', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'protein_id', 'ccdsid', 'ont']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: GTF file misses annotation for 317 genes in adata.\n",
      "WARNING: Skipped 90 genes because of duplicate identifiers in GTF file.\n"
     ]
    }
   ],
   "source": [
    "cnv.io.genomic_position_from_gtf('/home/jiehoonk/mnt/annotations/gencode.vM35.chr_patch_hapl_scaff.annotation.gtf', adata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... storing 'gene_id' as categorical\n",
      "... storing 'gene_name' as categorical\n",
      "... storing 'gene_id' as categorical\n",
      "... storing 'gene_name' as categorical\n",
      "... storing 'gene_id' as categorical\n",
      "... storing 'gene_name' as categorical\n",
      "... storing 'gene_id' as categorical\n",
      "... storing 'gene_name' as categorical\n"
     ]
    }
   ],
   "source": [
    "for query in [\"Mouse1\", \"Mouse2\", \"Mouse3\", \"Mouse4\"]:\n",
    "    bdata = adata[adata.obs.sample_id == query].copy()\n",
    "    bdata.write(f'./tmp/{query}.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "29009 features across 7866 samples within 1 assay \n",
       "Active assay: RNA (29009 features, 0 variable features)\n",
       " 2 layers present: counts, data\n",
       " 4 dimensional reductions calculated: Xpca_, XscANVI_, XscVI_, Xumap_"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query  <- \"Mouse1\"\n",
    "obj  <- schard::h5ad2seurat(glue('./tmp/', query, '.h5ad'))\n",
    "obj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \" raw data - genes: 29009 cells: 7866\"\n",
      "[1] \"1) Filter: cells > 200 genes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 1.7 GiB”\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 1.7 GiB”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2) Filter: genes > 10% of cells\"\n",
      "[1] \"10520 genes past filtering\"\n",
      "[1] \"3) Annotations gene coordinates\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "\n",
      "Attaching package: ‘foreach’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"found 30 confident non malignant cells\"\n",
      "[1] \"9406 genes annotated\"\n",
      "[1] \"4) Filter: genes involved in the cell cycle\"\n",
      "[1] \"8937 genes past filtering \"\n",
      "[1] \"5)  Filter: cells > 5genes per chromosome \"\n",
      "[1] \"6) Log Freeman Turkey transformation\"\n",
      "[1] \"A total of 7865 cells, 8937 genes after preprocessing\"\n",
      "[1] \"7) Measuring baselines (confident normal cells)\"\n",
      "[1] \"8) Smoothing data\"\n",
      "[1] \"9) Segmentation (VegaMC)\"\n",
      "[1] \"10) Adjust baseline\"\n",
      "[1] \"11) plot heatmap\"\n",
      "[1] \"found 6703 tumor cells\"\n",
      "[1] \"time classify tumor cells:  7.9722402493159\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: SingleCellExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse, colCounts, colCummaxs, colCummins,\n",
      "    colCumprods, colCumsums, colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs, colMads, colMaxs,\n",
      "    colMeans2, colMedians, colMins, colOrderStats, colProds, colQuantiles, colRanges, colRanks, colSdDiffs,\n",
      "    colSds, colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads, colWeightedMeans,\n",
      "    colWeightedMedians, colWeightedSds, colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods, rowCumsums, rowDiffs, rowIQRDiffs,\n",
      "    rowIQRs, rowLogSumExps, rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins, rowOrderStats,\n",
      "    rowProds, rowQuantiles, rowRanges, rowRanks, rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs,\n",
      "    rowVars, rowWeightedMads, rowWeightedMeans, rowWeightedMedians, rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated,\n",
      "    eval, evalq, Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, saveRDS,\n",
      "    setdiff, table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    second, second<-\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:glue’:\n",
      "\n",
      "    trim\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:lubridate’:\n",
      "\n",
      "    %within%\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:sp’:\n",
      "\n",
      "    %over%\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with 'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SummarizedExperiment’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "Loading required package: scuttle\n",
      "\n",
      "\n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    normalize, path, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    %--%, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    as_data_frame, groups, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    compose, simplify\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    crossing\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tibble’:\n",
      "\n",
      "    as_data_frame\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    components\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"found 7 subclones\"\n",
      "percentage_cells_subsclone_1 percentage_cells_subsclone_2 percentage_cells_subsclone_3 percentage_cells_subsclone_4 \n",
      "                  0.09667313                   0.10920483                   0.24048933                   0.04759063 \n",
      "percentage_cells_subsclone_5 percentage_cells_subsclone_6 percentage_cells_subsclone_7 \n",
      "                  0.25331941                   0.07742802                   0.17529464 \n",
      "[1] \"Segmentation of subclone :  1\"\n",
      "[1] \"Segmentation of subclone :  2\"\n",
      "[1] \"Segmentation of subclone :  3\"\n",
      "[1] \"Segmentation of subclone :  4\"\n",
      "[1] \"Segmentation of subclone :  5\"\n",
      "[1] \"Segmentation of subclone :  6\"\n",
      "[1] \"Segmentation of subclone :  7\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n",
      "Warning message in data.frame(..., check.names = FALSE):\n",
      "“row names were found from a short variable and have been discarded”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$Mouse1_subclone1\n",
      "   Chr     Start       End Alteration segm.mean\n",
      "45   6   3372257  71810710          2  0.308057\n",
      "10   2  38823257  69713516          1  0.154948\n",
      "17   2 147184083 153241426         -1 -0.116444\n",
      "25   3 116918256 159935833          1  0.117038\n",
      "29   4  41699989  44990734         -1 -0.126453\n",
      "42   5 112276691 115098790         -1 -0.142721\n",
      "49   6 135362931 149563329          1  0.119146\n",
      "68   8 120668222 123254348         -1 -0.136834\n",
      "78   9  95857597 103230444          1  0.124489\n",
      "81  10  26753421 115582448          1  0.125165\n",
      " [ reached 'max' / getOption(\"max.print\") -- omitted 4 rows ]\n",
      "\n",
      "$Mouse1_subclone2\n",
      "   Chr     Start       End Alteration segm.mean\n",
      "20   3  75516490  89176921          1  0.121615\n",
      "45   7 125467640 142061034         -1 -0.204257\n",
      "\n",
      "$Mouse1_subclone3\n",
      "   Chr    Start      End Alteration segm.mean\n",
      "60   9 26733728 37524148          1  0.161791\n",
      "\n",
      "$Mouse1_subclone4\n",
      "    Chr    Start       End Alteration segm.mean\n",
      "114  14 78849178 119115074          1  0.154119\n",
      "\n",
      "$Mouse1_subclone5\n",
      "   Chr    Start       End Alteration segm.mean\n",
      "33   5 90227190 105915818         -1 -0.102473\n",
      "56   8 70697739  77245012         -1 -0.111058\n",
      "60   9 25108130  35558528          1  0.157206\n",
      "62   9 49518336  57062805          1  0.112042\n",
      "64   9 70207350  86464950          1  0.125563\n",
      "\n",
      "$Mouse1_subclone7\n",
      "   Chr     Start       End Alteration segm.mean\n",
      "63  10 116018213 122945795         -1 -0.108951\n",
      "\n",
      "$Mouse1_clone\n",
      "    Chr     Start       End Alteration segm.mean\n",
      "34    4 132899213 138326307         -2 -0.241783\n",
      "6     1 167371966 172384099         -1 -0.184024\n",
      "54    9  87702800  95568023         -1 -0.115832\n",
      "115  15  89130585  96284495          1  0.168578\n",
      "\n",
      "$Mouse1_shareSubclone\n",
      "                 Chr     Start       End Alteration segm.mean      sh_sub Mean\n",
      "Mouse1_share1.1    1   3205901  34308661          1  0.149454       1-2-3    0\n",
      "Mouse1_share2.1    1   3205901  33742564          1  0.106173       2-1-3    0\n",
      "Mouse1_share1.5    1 156558786 166409863          1  0.153644         1-3    0\n",
      "Mouse1_share1.11   1 167371966 172384099         -2 -0.226392 1-3-4-5-6-7    0\n",
      "Mouse1_share4.1    1 167371966 171890718         -2 -0.216544 4-1-3-5-6-7    0\n",
      "Mouse1_share5.1    1 167371966 172298064         -1 -0.197565 5-1-3-4-6-7    0\n",
      "Mouse1_share6.1    1 167371966 172122330         -2 -0.198890 6-1-3-4-5-7    0\n",
      " [ reached 'max' / getOption(\"max.print\") -- omitted 72 rows ]\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in RColorBrewer::brewer.pal(n = 12, name = \"RdBu\"):\n",
      "“n too large, allowed maximum for palette RdBu is 11\n",
      "Returning the palette you asked for with that many colors\n",
      "”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "Warning message in annoteBandOncoHeat(res_proc$count_mtx_annot, diffSubcl, res_subclones$n_subclones, :\n",
      "“NAs introduced by coercion”\n",
      "\u001b[90mIf you use the ggtree package suite in published research, please cite the appropriate paper(s):\n",
      "\n",
      "Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for\n",
      "visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods\n",
      "in Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628\n",
      "\n",
      "LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR Jones, T Bradley, H Zhu, Y Guan, Y Jiang,\n",
      "G Yu. treeio: an R package for phylogenetic tree input and output with richly annotated and associated\n",
      "data. Molecular Biology and Evolution. 2020, 37(2):599-603. doi: 10.1093/molbev/msz240 \u001b[39m\n",
      "\n",
      "\n",
      "Attaching package: ‘tidytree’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    parent\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ape’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:tidytree’:\n",
      "\n",
      "    drop.tip, keep.tip\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    degree, edges, mst, ring\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    where\n",
      "\n",
      "\n",
      "ggtree v3.14.0 Learn more at https://yulab-smu.top/contribution-tree-data/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for\n",
      "visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods\n",
      "in Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628\n",
      "\n",
      "\n",
      "Attaching package: ‘ggtree’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ape’:\n",
      "\n",
      "    rotate\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Average angle change [1] 0.115441043064516\n",
      "\n",
      "Average angle change [2] 0.0324113849902199\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "  |==============================================================================================================| 100%\n",
      "\n",
      "[1] \"time subclones:  13.3138616323471\"\n"
     ]
    }
   ],
   "source": [
    "results <- pipelineCNA(obj[[\"RNA\"]]$counts, sample = query, organism = 'mouse', par_cores = 48)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
